lib/ibm_cloud_sdk_core/base_service.rb

```diff
@@ -61,6 +57,15 @@ def add_default_headers(headers: {})
       headers.each_pair { |k, v| @conn.default_options.headers.add(k, v) }
     end
 
+    def configure_service(service_name)
+      config = get_service_properties(service_name) if service_name
+
+      @service_url = config[:url] unless config.nil? || config[:url].nil?
```

Comment:
Looks like this might be an issue if a user supplies a service url to the constructor. The url would be overwritten if it's present in external config

Reply:
unless we decide to go ahead with the full service factory impl for the Ruby generator, we don't actually need to refactor the external config-related function into the new "configure_service" method.
If/when we need to do that, I think we can work out a way (between some updates to the ruby generator and some changes in the ruby core) to make sure that the precedence order is properly implemented:
1. user-supplied service url
2. externally-configured service url
3. default service url

But at this point, we probably don't need to worry about that.
So to be clear, I think for now, we can back out the changes that you have made re: configure_service()

Reply:
Per our discussion on slack, I'll keep the configure_service method but just keep it invoked through the ctor

Reply:
Did we settle on overrirding the `service_url` in case there is `URL` in `config`?

Reply:
@mamoonraja If a user passes in a service url value when constructing the service, we want that value to win and override any value that was possibly found within the external config.
The precedence is listed in my comment above.   This is consistent with the other languages.